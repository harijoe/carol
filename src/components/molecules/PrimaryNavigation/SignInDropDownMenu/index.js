import React, { PropTypes } from 'react'

import { DropDownMenu, SignInForm } from 'containers'

const SignInDropDownMenu = ({ location, ...props }) => {
  if (location && location.pathname && (location.pathname === '/login')) {
    return null
  }

  return (
    <DropDownMenu label="login">
      <SignInForm {...props} form="SignInFormDropDownMenu" />
    </DropDownMenu>
  )
}

SignInDropDownMenu.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string,
  }),
}

export default SignInDropDownMenu
