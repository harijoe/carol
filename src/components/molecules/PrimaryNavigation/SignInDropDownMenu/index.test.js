import React from 'react'
import { shallow } from 'enzyme'

import SignInDropDownMenu from './'

const wrap = (props = {}) => shallow(<SignInDropDownMenu {...props} />)

describe('user is not on login page', () => {
  it('renders a form', () => {
    const wrapper = wrap({ location: { pathname: '/' } })

    expect(wrapper.node.props.label).toEqual('login')
    expect(wrapper).toMatchSnapshot()
  })
})

describe('user is on login page', () => {
  it('renders null', () => {
    const wrapper = wrap({ location: { pathname: '/login' } })

    expect(wrapper.node).toBeNull()
  })
})
