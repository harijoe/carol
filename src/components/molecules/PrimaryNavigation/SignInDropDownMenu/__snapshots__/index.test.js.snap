exports[`test renders PopinMenuButton 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.button
    onClick={[Function]}
    type="button">
    Test
</styled.button>,
  "nodes": Array [
    <styled.button
      onClick={[Function]}
      type="button">
      Test
</styled.button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <PopinMenuButton
        onClick={[Function]}>
        Test
</PopinMenuButton>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": "Test",
          "onClick": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.button
          onClick={[Function]}
          type="button">
          Test
</styled.button>,
        "_debugID": 2,
        "_renderedOutput": <styled.button
          onClick={[Function]}
          type="button">
          Test
</styled.button>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PopinMenuButton
    onClick={[Function]}>
    Test
</PopinMenuButton>,
}
`;
